shipwrighttests/features/steps/install_steps.py

```diff
@@ -1,17 +1,91 @@
 from behave import given, then, when, step
+from kubernetes import client, config
+from kubernetes.client import configuration
+from kubernetes.config import ConfigException
+from resources import ResourcesModel
+from hamcrest import assert_that, equal_to
 
 @given(u'the Kubernetes cluster is available')
 def probe_cluster_available(context):
-	pass
+	"""Probe if the cluster is available
+
+	Args:
+		context (_type_): It holds the contextual information during the execution of tests. 
+		It is an object that can store user-defined data along with Python Behave-defined data, in context attributes
+
+	Raises:
+		config.ConfigException: When unable to load kubeconfig
+	"""	
+	try:
+		config.load_incluster_config()
+	except config.ConfigException:
+		try:
+			config.load_kube_config()
+			print("Active host is %s" % configuration.Configuration().host)
+		except config.ConfigException:
+			raise Exception("cluster probe failed")
 
 @then(u'"{deployment}" is deployed and "{state}" on "{namespace}" namespace')
 def probe_deployment_state(context, deployment, state, namespace):
-	pass
+	"""probe the state of deployments
+
+	Args:
+		context (_type_): It holds the contextual information during the execution of tests. 
+		It is an object that can store user-defined data along with Python Behave-defined data, in context attributes
+		deployment (str): name of the deployment
+		state (str): current state of the deployment
+		namespace (str): namespace in which the deployment is present
+
+	Raises:
+		Exception: Raises a exception when deployment is not ready
 
+	Returns:
+		bool: True based on deployment state
+	"""	
+	appsv1 = client.AppsV1Api()
+	deployment_state = appsv1.read_namespaced_deployment_status(deployment, namespace)
+	if deployment_state.status.unavailable_replicas is None:
```

Comment:
A good example of how to achieve this check is the following snippet:

```bash
 kubectl --namespace=tekton-pipelines rollout status deployment tekton-pipelines-controller
```

Let's investigate what's behind this `rollout status deployment` subcommand to understand where we need to fetch this information from.